includes/class-amp-story-post-type.php

```diff
@@ -1209,6 +1113,94 @@ static function ( $font ) use ( $fonts_url, $subsets ) {
 		return $fonts;
 	}
 
+	/**
+	 * Get list of google fonts from Google fonts file used in AMP Stories.
+	 *
+	 * @param array  $fonts Array of existing fonts.
+	 * @param string $file  Path to google font file.
+	 *
+	 * @return array $fonts
+	 */
+	public static function get_google_fonts( $fonts, $file ) {
+		if ( ! is_readable( $file ) ) {
+			return $fonts;
+		}
+		$file_content = file_get_contents( $file );  // phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents
+		$google_fonts = json_decode( $file_content, true );
+		if ( empty( $google_fonts ) ) {
+			return $fonts;
+		}
+		$existing_fonts = wp_list_pluck( $fonts, 'name' );
+
+		foreach ( $google_fonts as $font ) {
+			// If font already defined, then don't load from the google fonts list.
+			if ( in_array( $font['family'], $existing_fonts, true ) ) {
+				continue;
+			}
+			$gfont    = '';
+			$variants = array_intersect(
+				$font['variants'],
+				[
+					'regular',
+					'italic',
+					'700',
+					'700italic',
+				]
+			);
+
+			$variants = array_map(
+				function ( $variant ) {
+					$variant = str_replace( '0italic', '0i', $variant );
+					$variant = str_replace( 'regular', '400', $variant );
+					$variant = str_replace( 'italic', '400i', $variant );
+
+					return $variant;
+				},
+				$variants
+			);
+
+			if ( $variants ) {
+				$gfont = $font['family'] . ':' . implode( ',', $variants );
+			}
+			$fonts[] = [
+				'name'      => $font['family'],
+				'fallbacks' => (array) self::get_fallback( $font['category'] ),
+				'gfont'     => $gfont,
+			];
+		}
+
+		return $fonts;
+	}
+
+	/**
+	 * Helper method to lookup fallback font.
+	 *
+	 * @param string $category Google font category.
+	 *
+	 * @return string $fallback Fallback font.
+	 */
+	public static function get_fallback( $category ) {
```

Comment:
Hmm we should rename that to `get_font_fallback` to make it more clear what it is about.

Reply:
Fixed in https://github.com/ampproject/amp-wp/pull/3110/commits/014f71fefe9eee5533d4a6cc7ad3bb5f370572dd